### PR TITLE
ToS on public shares

### PR DESCRIPTION
**Pull request by daita**

### CODE REVIEW

File: appinfo/app.php [hunk 1]
```diff
@@ -19,7 +19,15 @@
...
 require_once __DIR__ . '/../vendor/autoload.php';
 
 $app = new \OCA\TermsOfService\AppInfo\Application();
+Util::addStyle('terms_of_service', 'overlay');
 $app->register();
+
+Util::addScript('terms_of_service', 'terms_of_service_public');
+
+
+Util::connectHook('OC_Filesystem', 'preSetup', $app, 'addStorageWrapper');
```
Comment (nickvergessen) [inline appinfo/app.php]: No modifications in this file

**Comment by daita** [comment]
To enable the ToS on public shares: 

```
./occ config:app:set --value '1' terms_of_service tos_on_public_shares
```

**Comment by skjnldsv** [comment]
@daita can't get it to work.
Compiled, enabled, ran the occ command, and opened a public share. Nothing pops up :woman_shrugging:

File: lib/Controller/SigningController.php [hunk 2]
```diff
@@ -79,6 +89,22 @@ public function signTerms(int $termId): JSONResponse {
 		return new JSONResponse();
 	}
 
+
+	/**
+	 * @PublicPage
+	 *
+	 * @param int $termId
+	 *
```
Comment (rullzer) [inline lib/Controller/SigningController.php]:
```suggestion
	 * @UseSession
```

**Comment by MorrisJobke** [comment]
So this one never made it into master 😭

**Comment by nickvergessen** [comment]
https://github.com/nextcloud/terms_of_service/pull/185

**Comment by daita** [comment]
oups, my bad

**Pull request merged**